ci_watson/plugin.py

```diff
@@ -67,12 +67,12 @@ def pytest_runtest_setup(item):
 
 
 @pytest.fixture(scope='function')
-def _jail(tmpdir):
+def _jail(tmp_path):
     """Perform test in a pristine temporary working directory."""
     old_dir = os.getcwd()
-    os.chdir(tmpdir.strpath)
+    os.chdir(tmp_path)
     try:
-        yield tmpdir.strpath
+        yield str(tmp_path)
```

Comment:
You may want to just yield the `tmp_path` instead of its string form?

That is what is done here

https://github.com/astropy/astroquery/blob/4d2311745752ffc0133e439053dd4046e4a8294b/astroquery/conftest.py#L70-L78

in a similar fixture I wrote based on `_jail`.

This would allow one to be sure one is `pathlib` compatible, though it might break a few things downstream.  Though perhaps the breaking is good?

Or one could leave this fixture unchanged and make a properly-named fixtured called `tmp_cwd` that can be used that yields a `pathlib.Path` object, and can be transitioned to at anytime by any package that uses ci-watson.

Reply:
Thanks for the question and comments.

I added the `str` to keep this consistent for the downstream (as you mentioned changing it to a `Path` might break things).

Given the naming (`_jail`) I would have assumed this was not intended for "public" use. So replacing it with a public fixture (or having the downstream libraries implement their own for their needs) seems reasonable.

To use `-p no:legacyapth` here (to keep this test suite updated) `_jail` would need to be updated (as is done in this PR).

Reply:
In this package, this fixture is also used for HSTCAL. So unless you feel like patching up hstcal and acstools test suite, backward compatibility is a must.

Reply:
Definitely not!  Let's keep backwards compatibility.  😅